java/fury-core/src/main/java/org/apache/fury/meta/Encoders.java

```diff
@@ -139,4 +134,32 @@ public static MetaString encodeFieldName(String fieldName) {
     return fieldMetaStringCache.computeIfAbsent(
         fieldName, k -> FIELD_NAME_ENCODER.encode(fieldName, fieldNameEncodings));
   }
+
+  public static MetaString encodePackage(String pkg) {
+    return pgkMetaStringCache.computeIfAbsent(pkg, k -> PACKAGE_ENCODER.encode(pkg, pkgEncodings));
+  }
+
+  public static MetaString encodeTypeName(String typeName) {
+    return typeMetaStringCache.computeIfAbsent(
+        typeName, k -> TYPE_NAME_ENCODER.encode(typeName, typeNameEncodings));
+  }
+
+  public static String decodeFieldName(long hashCode, byte[] bytes, MetaString.Encoding encoding) {
```

Comment:
Hashcode is an internal optimization, maybe we shouldn't make it pubic as an API for meta string encoder